packages/swapper/src/swappers/zrx/ZrxSwapper.ts

```diff
@@ -1,7 +1,21 @@
-import { Swapper, SwapperType } from '../../api'
-
+import { GetQuoteInput, Quote, Swapper, SwapperType } from '../../api'
+import { getZrxQuote } from './getQuote/getQuote'
+export class ZrxError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.message = `ZrxError:${message}`
+  }
+}
 export class ZrxSwapper implements Swapper {
   getType() {
     return SwapperType.Zrx
   }
+
+  /**
+   * Get a basic quote (rate) for the pair
+   * @param input
+   */
+  async getQuote(input: GetQuoteInput): Promise<Quote | undefined> {
+    return getZrxQuote(input)
```

Comment:
Does this need to be `await`ed?

Reply:
No return automatically awaits

Reply:
Well... That's not technically correct. This function simply returns a Promise. It doesn't wait for the Promise to be resolved before returning. The caller can await the Promise.

The async keyword tells the engine to wrap all return values in a Promise. In this case, the async keyword isn't necessary because the return value is already a Promise and await isn't called.

Reply:
When it's an async function It automatically resolves the returned value.   It returns a Promise.resolve instead of a promise

Reply:
wait or are you saying because the function is already resolving the promise it's unneeded as long as the return type is Promise<Quote> type?